resolver/src/resolver_handle/mod.rs

```diff
@@ -59,167 +59,152 @@ impl DnsHandle for BasicResolverHandle {
     }
 }
 
-/// A Resolver for DNS records.
-pub struct ResolverFuture {
-    config: ResolverConfig,
-    options: ResolverOpts,
-    pub(crate) client_cache: CachingClient<LookupEither<BasicResolverHandle, StandardConnection>>,
-    hosts: Option<Arc<Hosts>>,
+/// A handle for resolving DNS records.
+///
+/// Creating a `ResolverHandle` returns a new handle and a future that should
+/// be spawned on an executor to drive the background work. The lookup methods
+/// on `ResolverHandle` request lookups from the background task.
+///
+/// Note that the futures returned by a `ResolverHandle` and the corresponding
+/// background task need not be spawned on the same executor, or be in the same
+/// thread. Additionally, one background task may have any number of handles;
+/// calling `clone()` on a handle will create a new handle linked to the same
+/// background task.
+///
+/// The background task future will finish once all existing handles to it
+/// have been dropped.
+#[derive(Clone)]
+pub struct ResolverHandle {
+    request_tx: mpsc::UnboundedSender<Request>,
```

Comment:
I was hoping we wouldn’t need this, but I guess we do need it, because of the inner NameServerPool interface, right?